tools/json-api-client/src/transport/json.ts

```diff
@@ -136,7 +137,24 @@ function parseResponse<T>(response: Response): Promise<T> {
   } else if (response.status === 401) {
     throw new AuthenticationError(response)
   } else if (response.status >= 500) {
-    throw new ServerError(response)
+    const json = await response.json()
+    let errors: ErrorItem[]
```

Comment:
Very minor nit: This could be a const, and then the assignments in the if branches could be `push`es.

Reply:
I'd rather have the `const` assignment, with the change that its a ternary, then we get the benefit of `.map` and const variable.

Reply:
How do you feel about the change I made to extract this into a function. End result is a `const` variable.

Reply:
I would prefer something like this
```ts
 const errors = json.errors ? json.errors.map((e: any) => ({
        status: response.status,
        detail: e.detail,
      })) : [
        {
          status: response.status,
          detail: response.statusText,
        },
      ]
```

Reply:
I'm personally not a fan of complex ternary's over multiple lines...

Reply:
I just find that something like this in a function is a level of unneeded indirection vs a ternary, but i'd be cool with a function if you feel strongly about ternaries.

Reply:
I think that my lack of formatting on that snippet makes it harder to read than it needs to be too :sweat_smile:

Reply:
I feel like the function is worthwhile as it could be used on other response codes that we don't currently handle. e.g. `422 Unprocessable Entity`, `429 Too Many Requests`

Reply:
SGTM, lets factor this out into a function